libraries/Microsoft.Bot.Builder/Teams/TeamsActivityHandler.cs

```diff
@@ -738,6 +738,61 @@ protected virtual Task OnTeamsTeamUnarchivedAsync(TeamInfo teamInfo, ITurnContex
+        /// or threads to receive notice of cancellation.</param>
+        /// <returns>A task that represents the work queued to execute.</returns>
+        /// <remarks>
+        /// In a derived class, override this method to add logic that applies to all event activities.
+        /// </remarks>
+        protected override Task OnEventActivityAsync(ITurnContext<IEventActivity> turnContext, CancellationToken cancellationToken)
+        {
+            if (turnContext.Activity.ChannelId == Channels.Msteams)
+            {
+                switch (turnContext.Activity.Name)
+                {
+                    case "application/vnd.microsoft.meetingStart":
+                        return OnTeamsMeetingStartAsync(JObject.FromObject(turnContext.Activity.Value).ToObject<MeetingStartEventDetails>(), turnContext, cancellationToken);
+                    case "application/vnd.microsoft.meetingEnd":
+                        return OnTeamsMeetingEndAsync(JObject.FromObject(turnContext.Activity.Value).ToObject<MeetingEndEventDetails>(), turnContext, cancellationToken);
+                }
+            }
+
+            return base.OnEventActivityAsync(turnContext, cancellationToken);
+        }
+
+        /// <summary>
+        /// Invoked when a Teams Meeting Start event activity is received from the connector.
+        /// Override this in a derived class to provide logic for when a meeting is started.
+        /// </summary>
+        /// <param name="meeting">The details of the meeting.</param>
+        /// <param name="turnContext">A strongly-typed context object for this turn.</param>
+        /// <param name="cancellationToken">A cancellation token that can be used by other objects
+        /// or threads to receive notice of cancellation.</param>
+        /// <returns>A task that represents the work queued to execute.</returns>
+        protected virtual Task OnTeamsMeetingStartAsync(MeetingStartEventDetails meeting, ITurnContext<IEventActivity> turnContext, CancellationToken cancellationToken)
+        {
+            return Task.CompletedTask;
+        }
+
+        /// <summary>
+        /// Invoked when a Teams Meeting End event activity is received from the connector.
+        /// Override this in a derived class to provide logic for when a meeting is ended.
+        /// </summary>
+        /// <param name="meeting">The details of the meeting.</param>
+        /// <param name="turnContext">A strongly-typed context object for this turn.</param>
+        /// <param name="cancellationToken">A cancellation token that can be used by other objects
+        /// or threads to receive notice of cancellation.</param>
+        /// <returns>A task that represents the work queued to execute.</returns>
+        protected virtual Task OnTeamsMeetingEndAsync(MeetingEndEventDetails meeting, ITurnContext<IEventActivity> turnContext, CancellationToken cancellationToken)
+        {
+            return Task.CompletedTask;
+        }
+
```

Comment:
Note: There's a couple of things I'm not sure about with these:

1. Argument order. Some methods (like [`OnTeamsFileConsent`](https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder/Teams/TeamsActivityHandler.cs#L148)) have `turnContext` first, while others (like [`OnTeamsChannelDeleted`](https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder/Teams/TeamsActivityHandler.cs#L622)) have `turnContext` come after the method-specific arguments (like `channelInfo` and `teamsInfo`).
2. Based on the other overrides and virtuals, it looks like not marking these methods as `async` is the right call. Just want to confirm.